WGLC updates 4/x (explicit typing discussion) (#476)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

## Explicit Typing {#explicit_typing}

[@RFC8725, section 3.11] describes the use of explicit typing as one mechanism to prevent confusion attacks
(described in [@RFC8725, section 2.8]) in which one kind of JWT is mistaken for another. SD-JWTs are also potentially

> subject to such confusion attacks, so in the absence of other techniques, it is RECOMMENDED that applications/profiles of SD-JWT specify an explicit type

Reviewer: ```suggestion
subject to such confusion attacks, so in the absence of other techniques, it is RECOMMENDED that applications profiles of SD-JWT specify an explicit type
```

Author: ```suggestion
subject to such confusion attacks, so in the absence of other techniques, it is RECOMMENDED that application profiles of SD-JWT specify an explicit type
```

by including the `typ` header parameter when the SD-JWT is issued, and for Verifiers to check this value.

When explicit typing using the `typ` header is employed for an SD-JWT, it is RECOMMENDED that a media type name of the format
"application/example+sd-jwt" be used, where "example" is replaced by the identifier for the specific kind of SD-JWT.
The definition of `typ` in Section 4.1.9 of [@!RFC7515] recommends that the "application/" prefix be omitted, so
"example+sd-jwt" would be the value of the `typ` header parameter.

> Use of the `cty` content type header parameter to indicate the content type of the SD-JWT payload is also a completely reasonable approach to indicating the specific kind of SD-JWT in question.

Reviewer: ```suggestion
The `cty` content type header parameter can also be used distinguish different types of JSON object, or different kinds of JWT claim set which are secured by the SD-JWT payload.
```
